routes/api/lists.js

```diff
@@ -1,6 +1,142 @@
 const express = require("express");
 const router = express.Router();
 
-router.get("/test", (req, res) => res.json({ msg: "This is the lists route" }));
+const List = require("../../models/List");
+const TextString = require("../../models/TextString");
+const ListVersion = require("../../models/ListVersion");
+
+// Show
+router.get("/:id", (req, res) => {
+  List.findById(req.params.id)
+    .populate("name")
+    .populate("currentVersionId")
+    .populate("description")
+    .then((list) => {
+      if (list) {
+        res.json(list);
+      } else {
+        res.status(404).json("No Such List");
+      }
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Index
+router.get("/", (req, res) => {
+  List.find({})
+    .populate("name")
+    .populate("currentVersionId")
+    .populate("description")
+    .then((lists) => {
+      res.json(lists);
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Create
+router.post("/", async (req, res) => {
+  let { name, description, currentVersionId, image } = req.body;
+
+  name = await TextString.create(name);
+  description = await TextString.create(description);
+
+  console.log(currentVersionId)
+  currentVersionId = await ListVersion.create(currentVersionId);
+
+  let list = new List({
+    name,
+    description,
+    currentVersionId,
+    image
+  });
+
+  list.currentVersionId.listId = list.id; 
+
+  if (image) {
+    list.image.data = image;
+    list.image.contentType = "image/png";
```

Comment:
Since this is repeating, add a `utils/image.js` file, and add a function:
```
function getImageObject(imageData) {
  return {
    data: imageData,
    contentType: "image/png"
  };
}
````
then do: 
`list.image = utils.image.getImageObject(image)`